File: src/api/schemas.py
"""Pydantic schemas for API request/response models."""
from pydantic import BaseModel, Field
from uuid import UUID
from typing import List


class ProductCreate(BaseModel):
    """Schema for creating a new product."""
    name: str = Field(..., min_length=1, description="Product name")
    price: float = Field(..., ge=0, description="Product price (must be non-negative)")
    stock: int = Field(..., ge=0, description="Product stock quantity")


class ProductResponse(BaseModel):
    """Schema for product response."""
    id: str
    name: str
    price: float
    stock: int

    class Config:
        from_attributes = True


class UpdateStock(BaseModel):
    """Schema for updating product stock."""
    quantity: int = Field(..., description="Quantity to add (positive) or remove (negative)")


class ErrorResponse(BaseModel):
    """Schema for error responses."""
    detail: str

File: backend/typescript/src/api/schemas.ts
import { z } from "zod";

export const productCreateSchema = z.object({
  name: z.string().min(1),
  price: z.number().min(0),
  stock: z.number().int().min(0),
});

export const updateStockSchema = z.object({
  quantity: z.number().int(),
});

export const productResponseSchema = z.object({
  id: z.string(),
  name: z.string(),
  price: z.number(),
  stock: z.number(),
});

export const errorResponseSchema = z.object({
  detail: z.string(),
});

export type ProductCreate = z.infer<typeof productCreateSchema>;
export type UpdateStock = z.infer<typeof updateStockSchema>;
export type ProductResponse = z.infer<typeof productResponseSchema>;
export type ErrorResponse = z.infer<typeof errorResponseSchema>;
